internal/pkg/aws/ec2/ec2.go

```diff
@@ -186,23 +161,65 @@ func (c *EC2) HasDNSSupport(vpcID string) (bool, error) {
 	return aws.BoolValue(resp.EnableDnsSupport.Value), nil
 }
 
-// ListVPCSubnets lists all subnets given a VPC ID.
-func (c *EC2) ListVPCSubnets(vpcID string, opts ...ListVPCSubnetsOpts) ([]string, error) {
-	respSubnets, err := c.subnets(Filter{
+// VPCSubnets are all subnets within a VPC.
+type VPCSubnets struct {
+	Public  []VPCResource
+	Private []VPCResource
+}
```

Comment:
hmm following the logic here https://github.com/aws/copilot-cli/pull/2457#discussion_r654026948 i think it might be clearer to just have
```go
// VPCSubnets are all subnets within a VPC.
type VPCSubnets struct {
	Public  []Subnet
	Private []Subnet
}
```